doc/chplvis/chplvis.rst

```diff
@@ -0,0 +1,540 @@
+execution.   This list gives details about the *gets*, *puts* and
+*forks* initiated by this task.
+
+In the main window, clicking on a communication line will create a
+window with communication information for that link.  Clicking red
+part of the line between locale 0 and locale 1 will produce a window
+that looks like:
+
+.. image:: E1-C1.0.png
+
+It is important to notice the direction of the *arrow* in the header
+for the windows.  This is for communication from locale 1 to locale 0.
+The total number of communication calls was 12.  It is further broken
+out into three components:
+
+  - *Gets:* This is a communication call initiated by locale 0 to
+    get a data located on locale 1.
+
+  - *Puts:* This is a communication call initiated by *locale 1*
+    to put data from locale 1 onto locale 0.
+
+  - *Forks:* This where *locale 1* starts a task running
+    on locale 0.  As part of the task start, a block of data is sent to
+    locale 0 as an argument to the task.  This data is considered a
+    communication call  by *chplvis*.
+
+.. _`Example 2`: 
+    
+Example 2
+---------
+
+In many programs, one will want to look at a number of small parts of
+their program in addition to seeing the total statistics.  prog2.chpl
+gives an example of using the *VisualDebug* functions
+*tagVdebug(name)* and *pauseVdebug()*.
+
+::
+
+    // Example 2 of use of VisualDebug module and chplvis tool.
+
+    use BlockDist;
+    use VisualDebug;
+
+    config var ncells = 10;
+
+    proc main() {
+
+       // Create a couple of domains and a block mapped data array.
+       const Domain = { 1 .. ncells };
+       const mapDomain = Domain dmapped Block(Domain);
+
+       var  data : [mapDomain] int = 1;
+
+       // Start VisualDebug here
+       startVdebug ("E2");
+
+       // First computation step ... a simple forall
+       forall i in Domain do data[i] += here.id + 1;
+
+       // Write the result, we want to see the results of the above
+       // so we tag before we continue.
+       tagVdebug("writeln 1");
+       writeln("data= ", data);
+    
+       // Second computation step ... using the distributed domain
+       tagVdebug("step 2");
+       forall i in mapDomain do data[i] += here.id+1;
+
+       // Don't capture the writeln
+       pauseVdebug();
+       writeln("data2= ", data);
+
+       // Reduction step
+       tagVdebug("reduce");
+       var i = + reduce data;
+
+       // done with visual debug
+       stopVdebug();
+
+       writeln ("sum is " + i + ".");
+    }
+
+
+Note that the *startVdebug("E2")* is placed after the declarations
+so that tasks and communication for the declarations are not included.
+The initial display of *chplvis* shows data for the entire run. (This
+program was run on five locales.)
+
+.. image:: E2-1.png
+
+There is now a new menu called *Tags* that reflects the
+*tagVdebug()* calls in the program.  Selecting the tags menu
+gives the following display:
+
+.. image:: E2-2.png
```

Comment:
Does this image need to be updated?  The numbering for the tags seems different than a later image.